### PR TITLE
Changes for sai hostintf rx and tx with callback.

### PR DESCRIPTION
**Pull request by pierce-m**

### CODE REVIEW

File: switchapi/src/switch_hostif.c [hunk 1]
```diff
@@ -739,9 +743,10 @@ switch_api_hostif_rx_packet_from_hw(switch_packet_header_t *packet_header, char
 
     JLG(temp, switch_hostif_rcode_array, cpu_header->reason_code);
     if (!temp) {
-        SWITCH_API_ERROR("rx_packet w/ un-handled reason_code 0x%x\n",
-                            cpu_header->reason_code);
-        return SWITCH_STATUS_ITEM_NOT_FOUND;
+        JLG(temp, switch_hostif_rcode_array, SWITCH_HOSTIF_REASON_CODE_NONE);
+       // SWITCH_API_ERROR("rx_packet w/ un-handled reason_code 0x%x\n",
+       //                     cpu_header->reason_code);
+       // return SWITCH_STATUS_ITEM_NOT_FOUND;
```
Comment (pierce-m) [inline switchapi/src/switch_hostif.c]: Any advice on how to handle this correctly?

File: switchsai/src/saivlan.c [hunk 2]
```diff
@@ -57,6 +58,38 @@ sai_status_t sai_create_vlan_entry(
         switch_status = switch_api_vlan_mld_snooping_enabled_set(
             vlan_handle, true);
         assert(switch_status == SWITCH_STATUS_SUCCESS);
+
+        /* register rx net filter for this vlan */
```
Comment (srikrishnagopu) [inline switchsai/src/saivlan.c]: is not this rule specific to fboss ? should we move this up or add it as user defined attribute from SAI for fboss alone ?

File: switchapi/src/switch_packet.c [hunk 3]
```diff
@@ -214,46 +212,80 @@ switch_packet_rx_to_host(
...
-            memcpy(in_packet, packet, packet_size);
+            memcpy(transformed_packet, packet, *packet_size);
         }
     } else {
-        memcpy(in_packet, packet, packet_size);
+        memcpy(transformed_packet, packet, *packet_size);
+    }
+}
+
+void
```
Comment (srikrishnagopu) [inline switchapi/src/switch_packet.c]: same as above. switch_packet_rx_filter_send ?

File: switchapi/src/switch_hostif.c [hunk 4]
```diff
@@ -801,8 +814,15 @@ switch_api_hostif_tx_packet(switch_device_t device, switch_hostif_packet_t *host
...
     cpu_header->tx_bypass = hostif_packet->tx_bypass;
-    cpu_header->reason_code = 0xFFFF;
-    switch_packet_tx_to_hw(&packet_header, hostif_packet->pkt, hostif_packet->pkt_size);
+    if (cpu_header->tx_bypass) {
+        cpu_header->reason_code = 0xFFFF;
+        switch_packet_tx_to_hw(&packet_header, hostif_packet->pkt,
+            hostif_packet->pkt_size);
+    } else {
+        cpu_header->reason_code = 0;
+        switch_packet_tx_switched(&packet_header,
```
Comment (srikrishnagopu) [inline switchapi/src/switch_hostif.c]: can we change the name here to something like switch_packet_tx_filter_send ?

File: switchapi/src/switch_hostif.c (hunk 1 above)
Comment (srikrishnagopu) [inline switchapi/src/switch_hostif.c]: Ideally any non zero reason code has to be registered in order to receive the packet. For now, remove the changes here.

File: switchsai/src/saivlan.c (hunk 2 above)
Comment (pierce-m) [inline switchsai/src/saivlan.c]: What do you think? I could move it up, but it makes the SAI stuff in FBOSS less portable. But maybe it already isn't portable...

Comment (pierce-m) [inline switchsai/src/saivlan.c]: I'm going to move it up for now, we can come back to the idea of adding a custom attribute for this.

Comment (srikrishnagopu) [inline switchsai/src/saivlan.c]: yes move it in fboss layer.

File: switchapi/src/switch_hostif.c (hunk 4 above)
Comment (pierce-m) [inline switchapi/src/switch_hostif.c]: How about tx_to_filter

**Pull request merged**